### PR TITLE
libbpf-tools: add xfsdist

### PR DESCRIPTION
**Pull request by chenhengqi**
```shell
$ sudo ./xfsdist -T -p 9891 1 1000
Tracing xfs operation latency... Hit Ctrl-C to end.

22:55:01
operation = 'write_iter'
     usecs               : count    distribution
         0 -> 1          : 381683   |****************************************|
         2 -> 3          : 279      |                                        |
         4 -> 7          : 42       |                                        |
         8 -> 15         : 9        |                                        |
        16 -> 31         : 3        |                                        |
        32 -> 63         : 1        |                                        |


22:55:02
operation = 'write_iter'
     usecs               : count    distribution
         0 -> 1          : 317963   |****************************************|
         2 -> 3          : 239      |                                        |
         4 -> 7          : 22       |                                        |
         8 -> 15         : 5        |                                        |
        16 -> 31         : 0        |                                        |
        32 -> 63         : 1        |                                        |
```

Signed-off-by: Hengqi Chen <chenhengqi@outlook.com>

### CODE REVIEW

File: libbpf-tools/xfsdist.c [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+		}
+		pos_args++;
+		break;
+	default:
+		return ARGP_ERR_UNKNOWN;
+	}
+	return 0;
+}
+
+int libbpf_print_fn(enum libbpf_print_level level,
```
Comment (anakryiko) [inline libbpf-tools/xfsdist.c:104]: static

File: libbpf-tools/xfsdist.bpf.c [hunk 2]
```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2021 Hengqi Chen
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include "bits.bpf.h"
+#include "xfsdist.h"
+
+#define MAX_ENTRIES		10240
+#define lock_xadd(ptr, val)	((void)__sync_fetch_and_add(ptr, val))
```
[the remark this answers is not in the document]
Comment (chenhengqi) [reply libbpf-tools/xfsdist.bpf.c]: Nope, personal taste.

File: libbpf-tools/xfsdist.bpf.c [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	return probe_entry();
+}
+
+SEC("fexit/xfs_file_fsync")
+int BPF_PROG(xfs_file_fsync_fexit)
+{
+	return probe_return(FSYNC);
+}
+
+char LICENSE[] SEC("license") = "GPL";
```
[the remark this answers is not in the document]
Comment (chenhengqi) [reply libbpf-tools/xfsdist.bpf.c:169]: Ah, I believe all libbpf tools (except execsnoop) in this repo derive their SPDX from your first tool `runqslower`. And I follow this pattern.

In libbpf-bootstrap, I have seen dual license.

Could you please elaborate more on this ?

All I know is that we need license section to satisfy the verifier and for some bpf helpers. (NOT quite sure.)

[the remark this answers is not in the document]
Comment (anakryiko) [reply libbpf-tools/xfsdist.bpf.c:169]: LGPL + BSD dual license allows to use parts of code for non-GPL code, if the user chooses to, as far as I understand. And once you use such dual-licensing, "Dual BSD/GPL" license value here makes more sense. Both are accepted by BPF verifier as GPL-compatible and all the same sets of helpers are supported for both.

[the remark this answers is not in the document]
Comment (chenhengqi) [reply libbpf-tools/xfsdist.bpf.c:169]: Ah, thanks. Will do in this way.

**Pull request closed without merging**